### PR TITLE
Add increment + decrement routes

### PR DESCRIPTION
**Pull request by abnegate**
## What does this PR do?

(Provide a description of what this PR does and why it's needed.)

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Screenshots may also be helpful.)

## Related PRs and Issues

- (Related PR or issue)

## Checklist

- [ ] Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
- [ ] If the PR includes a change to an API's metadata (desc, label, params, etc.), does it also include updated API specs and example docs?

## Summary by CodeRabbit

- **New Features**
  - Added API endpoints to increment and decrement numeric attributes of a document directly.
- **Tests**
  - Introduced comprehensive tests for the new increment and decrement attribute API endpoints, including validation of boundary conditions and error handling.

### CODE REVIEW

File: app/controllers/api/databases.php [hunk 1]
```diff
@@ -4462,6 +4464,160 @@ function updateAttribute(
...
+            throw new Exception(Exception::DATABASE_NOT_FOUND);
+        }
+
+        $collection = Authorization::skip(fn () => $dbForProject->getDocument('database_' . $database->getSequence(), $collectionId));
+        if ($collection->isEmpty()) {
+            throw new Exception(Exception::COLLECTION_NOT_FOUND);
+        }
+
+        try {
+            $document = $dbForProject->decreaseDocumentAttribute(
```
Comment (eldadfux) [inline app/controllers/api/databases.php:4591]: Should we also do some extra type check for arrays? what will happen when used on an array?

Comment (abnegate) [reply app/controllers/api/databases.php:4591]: We'll throw an exception, handled in the library
